cupy/cuda/graph.pyx

```diff
@@ -78,3 +77,25 @@ cdef class Graph:
         else:
             stream_ptr = stream.ptr
         runtime.graphUpload(self.graphExec, stream_ptr)
+
+    cpdef debug_dot_str(self, flags=0):
```

Comment:
nit: 
```suggestion
    cpdef str debug_dot_str(self, flags=0):
```

btw since this is a dot file, perhaps we can optionally support building a NetworkX graph out of it?

Reply:
> perhaps we can optionally support building a NetworkX graph out of it?

I've confirmed `nx.nx_agraph.read_dot()` can read the output! But I'd prefer to leave the diagnosis part to users. I think most users troubleshooting CUDA graph will try to visualize it (using their preferred tools, e.g. [GraphViz Online](https://dreampuf.github.io/GraphvizOnline/?compressed=CYSw5gTghgDgFgAmAewC4IN4CgDOBXAI0lkQGMAbPHVAUwgH0BGTLcqAm8gXgCJj4mPBPzgBtagE9yNXsCg44NYDwC6Abix9oAxvQB2yYDXoAGHuNRSZPAsnLKEC2NYg1SyCA7YduPDAGkAUQAlADlAgBksAB9MAEkAEQRYkwQAClRkGGQ44AAuBAAmAEpkhHoALQAOAGs6PU56AEEANwAdAB5OzsYAGj6TNoA%2BYeGAXxjMDAMjBDgoPWBpMoAzPD1SOYWlmjGyjBMADxMTkwBWM4A2QqaEgE47xgB2QtSU49OL69u7qsumgDCJjGE1iGCgpFINBwOAACnYQKQJAB1ECLZAAd32BHkxhgqAgZT0eAAtvQCBJaDgynAQKhglBUCBkDS6bCIFkyiSQDD2Vk9mCjqdhSLhWU3ggTAA6UVlUJykGTDDuLJ0RkgFo0cWgzAwCDM-WWbVYMaqDSaERMfSGYyMcySaS8HAIhxOGAuNyoBZgaRCbycXjMDJZHL5BCMYpYGZa%2BaLaQFIWfK43e6PAFnExYAAS23j4awABUJO6CnEAGJYS2FTTqLAW7RwK3R%2BiFe2WR02Oyu%2Bbu3iudyeP3sAN%2BIJhSJKxJlQrpTLZXIJ0qxSq1eqNABC7S6276A1GIyGOow0xtWzjWtiaw2Z52AswiZOXxTgSaABZCuuJQ-zsmfk9AmcjDAkeEJQryCJIqi6JYmCOI4HiBJEqS5KUtCrL0uqLKxLSqB8jAXI8nCHIwHeBwfKKFFfuKMoinKCpHiq7rQEymrGmCeoGnSEjGqatb1iQTY2qYQgALRDAgWgCbozZ2ggoiKFASzDtwJi1pJOjWkYggIGJEmWtJQmtnJClKT4XCqRoEz4EQDYIBQVC0AwM7YP6vhVkIIgWFYsjyIoyhqVWmnGGYXkds65AgN2zh9m4HheMpvDYAAsoESUArCACaSoYFOsTRje0ikakwbzmGADMS6SuRP7fAkL6FK%2BQKKmCNRosAZRZpkSRpFmADyADKBYILCTQAOIvuuESBAgmQIHVABqcQAoExRHhgOAQKQsKIbEwDUNtECkTAdKkIgsT4oSsSHDgIAAF4XggEg3fdZTHagp2vTtCDXXdD1Pb9RXitVT63K%2BVTrlUJzitDKRA8KTxlncL4ApDYqwykzVTBtW3INSYIYIc2r7NxGP7Ld2ogvse24bj%2BwE0TYIk5KpHkxjlNgoEhy0Ho6D46iwCoGdCBPJcVRVKRWY0OAcC8%2BGpEJDQ%2BJC4wIKKrx5rqY2hRBYIoXWLY9hCG6HoDvFZmjiE4RRGCuXhrOIYLkUlWVIwjB6BAJLa3UEANOQuTAMAEQgCYkTB6HjCFFUgSBC0jCXKQhBsDgoQe4UsJQNAJI4HEIAFvQBa6IE3TbnclwmP0ke9IMB7jNl%2BWxjsqzrJsDeFfs34g3VTQ3E8X7A7%2BCQJP%2BlxNSBkLQnCEEom1mLYri9AXUhZIUlS6EMkyWFzGyxEEbyxFFdVlFo5K1GyrE8rn5jyrIKqzEag9wJKhxHhcTxZp1prLY662eu8AbUW9h4P2OKQ5zYBEthOG2SRYgzhKqGAoEYyguzdqneg3tfb%2B0DmHIOIcg4RyjjHOOCcCBJxTp7dOmds653zoXYuPQK5VCrvuWu%2BN665geleFu7CD7Ck7vVMsgQ%2B68IHkPQII9gJKlAhPeEEVIIzxgpgOCCFLoIGJMvVCeMt4YQ3uhPCu8iL8nbofI%2BUN0aShosfC%2BCBQhX0YmqFiD8jzP0NEzR%2B6sP6BWbGVNs3keDhUikbHsJsQEIFcolLAKU0qZWyrbNh55SJlXtqVRccMky1RfGVC4EiWptTKAkLq6QFpLWmrNIpy1VrZWxgdMo1MDpHROkLReV1np-Raa9Bpn0VE-RerEf691AbvHhojZGkNGCCKomYoRpwEZI3XAkJoJhXimJPmzSpm14SaOPITUmjMGaYFZszdmmBam03xts5mxM9kYAOcBUinNuay2PPzQWZRxb7EltLWWKt9gKyVmUFWqsTTv34gIbWzYzA6XEp-MFQlZLyRoIpMJFkQVax1rJXS0Lv5CHhYihKyLMXNiMhizxQlvHGQRaZAMyKJhjCAA#digraph%20dot%20%7B%0Asubgraph%20cluster_1%20%7B%0Alabel%3D%22graph_1%22%20graph%5Bstyle%3D%22dashed%22%5D%3B%0A%22graph_1_node_0%22%5Bstyle%3D%22bold%22%20shape%3D%22record%22%20label%3D%22%7BKERNEL%0A%7C%20%7BID%20%7C%200%20(topoId%3A%202)%20%7C%20_Z8kernel_Av%5C%3C%5C%3C%5C%3C1%2C1%2C0%5C%3E%5C%3E%5C%3E%7D%0A%7C%20%7B%7Bnode%20handle%20%7C%20func%20handle%7D%20%7C%20%7B0x00005562AD991720%20%7C%200x00005562AD986AC0%7D%7D%0A%7C%20%7BaccessPolicyWindow%20%7C%20%7Bbase_ptr%20%7C%20num_bytes%20%7C%20hitRatio%20%7C%20hitProp%20%7C%20missProp%7D%20%7C%20%7B0x0000000000000000%20%7C%200%20%7C%200.000000%20%7C%20N%20%7C%20N%7D%7D%0A%7C%20%7Bcooperative%20%7C%200%7D%0A%7C%20%7Bpriority%20%7C%200%7D%0A%7D%22%5D%3B%0A%0A%22graph_1_node_1%22%5Bstyle%3D%22solid%22%20shape%3D%22rectangle%22%20label%3D%221%20(topoId%3A%201)%0Anode%20handle%3A%200x00005562AD991C50%0AHandle%3A%201%0AType%3A%20IF%0Agraph_2%0A%22%5D%3B%0A%0A%22graph_1_node_2%22%5Bstyle%3D%22bold%22%20shape%3D%22record%22%20label%3D%22%7BKERNEL%0A%7C%20%7BID%20%7C%202%20(topoId%3A%200)%20%7C%20_Z8kernel_Bv%5C%3C%5C%3C%5C%3C1%2C1%2C0%5C%3E%5C%3E%5C%3E%7D%0A%7C%20%7B%7Bnode%20handle%20%7C%20func%20handle%7D%20%7C%20%7B0x00005562ADEA42B0%20%7C%200x00005562AD97E510%7D%7D%0A%7C%20%7BaccessPolicyWindow%20%7C%20%7Bbase_ptr%20%7C%20num_bytes%20%7C%20hitRatio%20%7C%20hitProp%20%7C%20missProp%7D%20%7C%20%7B0x0000000000000000%20%7C%200%20%7C%200.000000%20%7C%20N%20%7C%20N%7D%7D%0A%7C%20%7Bcooperative%20%7C%200%7D%0A%7C%20%7Bpriority%20%7C%200%7D%0A%7D%22%5D%3B%0A%0A%22graph_1_node_0%22%20-%3E%20%22graph_1_node_1%22%20%5Bheadlabel%3D0%5D%3B%0A%22graph_1_node_1%22%20-%3E%20%22graph_1_node_2%22%20%5Bheadlabel%3D0%5D%3B%0A%7D%0Asubgraph%20cluster_2%20%7B%0Alabel%3D%22graph_2%22%20graph%5Bstyle%3D%22dashed%22%5D%3B%0A%22graph_2_node_0%22%5Bstyle%3D%22solid%22%20shape%3D%22record%22%20label%3D%22%7B%0AMEMCPY%0A%7C%20%7B%7BID%20%7C%20node%20handle%7D%20%7C%20%7B0%20(topoId%3A%203)%20%7C%200x00005562ADEA24C0%7D%7D%0A%7C%20%7Bkind%20%7C%20HtoD%20(HOST%20PAGEABLE%20to%20DEVICE)%7D%0A%7C%20%7B%7BsrcPtr%20%7C%20dstPtr%7D%20%7C%20%7Bpitch%20%7C%20ptr%20%7C%20xsize%20%7C%20ysize%20%7C%20pitch%20%7C%20ptr%20%7C%20xsize%20%7C%20ysize%7D%20%7C%20%7B0%20%7C%200x00005562AD48B800%20%7C%200%20%7C%200%20%7C%200%20%7C%200x00007F9EAC800000%20%7C%200%20%7C%200%7D%7D%0A%7C%20%7B%7BsrcPos%20%7C%20%7B%7Bx%20%7C%200%7D%20%7C%20%7By%20%7C%200%7D%20%7C%20%7Bz%20%7C%200%7D%7D%7D%20%7C%20%7BdstPos%20%7C%20%7B%7Bx%20%7C%200%7D%20%7C%20%7By%20%7C%200%7D%20%7C%20%7Bz%20%7C%200%7D%7D%7D%20%7C%20%7BExtent%20%7C%20%7B%7BWidth%20%7C%207688%7D%20%7C%20%7BHeight%20%7C%201%7D%20%7C%20%7BDepth%20%7C%201%7D%7D%7D%7D%0A%7D%22%5D%3B%0A%0A%22graph_2_node_1%22%5Bstyle%3D%22bold%22%20shape%3D%22record%22%20label%3D%22%7BKERNEL%0A%7C%20%7BID%20%7C%201%20(topoId%3A%202)%20%7C%20_Z11nrm2_kernelIdddLi0ELi0ELi128EEv16cublasNrm2ParamsIiT_T1_E%5C%3C%5C%3C%5C%3C960%2C128%2C0%5C%3E%5C%3E%5C%3E%7D%0A%7C%20%7B%7Bnode%20handle%20%7C%20func%20handle%7D%20%7C%20%7B0x00005562ADEA2A70%20%7C%200x00005562ADD7E6C0%7D%7D%0A%7C%20%7BaccessPolicyWindow%20%7C%20%7Bbase_ptr%20%7C%20num_bytes%20%7C%20hitRatio%20%7C%20hitProp%20%7C%20missProp%7D%20%7C%20%7B0x0000000000000000%20%7C%200%20%7C%200.000000%20%7C%20N%20%7C%20N%7D%7D%0A%7C%20%7Bcooperative%20%7C%200%7D%0A%7C%20%7Bpriority%20%7C%200%7D%0A%7D%22%5D%3B%0A%0A%22graph_2_node_2%22%5Bstyle%3D%22bold%22%20shape%3D%22record%22%20label%3D%22%7BKERNEL%0A%7C%20%7BID%20%7C%202%20(topoId%3A%201)%20%7C%20_Z11nrm2_kernelIdddLi0ELi0ELi128EEv16cublasNrm2ParamsIiT_T1_E%5C%3C%5C%3C%5C%3C1%2C128%2C0%5C%3E%5C%3E%5C%3E%7D%0A%7C%20%7B%7Bnode%20handle%20%7C%20func%20handle%7D%20%7C%20%7B0x00005562ADEA2FE0%20%7C%200x00005562ADD7E6C0%7D%7D%0A%7C%20%7BaccessPolicyWindow%20%7C%20%7Bbase_ptr%20%7C%20num_bytes%20%7C%20hitRatio%20%7C%20hitProp%20%7C%20missProp%7D%20%7C%20%7B0x0000000000000000%20%7C%200%20%7C%200.000000%20%7C%20N%20%7C%20N%7D%7D%0A%7C%20%7Bcooperative%20%7C%200%7D%0A%7C%20%7Bpriority%20%7C%200%7D%0A%7D%22%5D%3B%0A%0A%22graph_2_node_3%22%5Bstyle%3D%22solid%22%20shape%3D%22record%22%20label%3D%22%7B%0AMEMCPY%0A%7C%20%7B%7BID%20%7C%20node%20handle%7D%20%7C%20%7B3%20(topoId%3A%200)%20%7C%200x00005562ADEA3550%7D%7D%0A%7C%20%7Bkind%20%7C%20DtoD%20(DEVICE%20to%20DEVICE)%7D%0A%7C%20%7B%7BsrcPtr%20%7C%20dstPtr%7D%20%7C%20%7Bpitch%20%7C%20ptr%20%7C%20xsize%20%7C%20ysize%20%7C%20pitch%20%7C%20ptr%20%7C%20xsize%20%7C%20ysize%7D%20%7C%20%7B0%20%7C%200x00007F9EAC801E00%20%7C%200%20%7C%200%20%7C%200%20%7C%200x00007F9EBDA02000%20%7C%200%20%7C%200%7D%7D%0A%7C%20%7B%7BsrcPos%20%7C%20%7B%7Bx%20%7C%200%7D%20%7C%20%7By%20%7C%200%7D%20%7C%20%7Bz%20%7C%200%7D%7D%7D%20%7C%20%7BdstPos%20%7C%20%7B%7Bx%20%7C%200%7D%20%7C%20%7By%20%7C%200%7D%20%7C%20%7Bz%20%7C%200%7D%7D%7D%20%7C%20%7BExtent%20%7C%20%7B%7BWidth%20%7C%208%7D%20%7C%20%7BHeight%20%7C%201%7D%20%7C%20%7BDepth%20%7C%201%7D%7D%7D%7D%0A%7D%22%5D%3B%0A%0A%22graph_2_node_0%22%20-%3E%20%22graph_2_node_1%22%20%5Bheadlabel%3D0%5D%3B%0A%22graph_2_node_1%22%20-%3E%20%22graph_2_node_2%22%20%5Bheadlabel%3D0%5D%3B%0A%22graph_2_node_2%22%20-%3E%20%22graph_2_node_3%22%20%5Bheadlabel%3D0%5D%3B%0A%7D%0A%7D)) rather than traversing it within Python console.